### PR TITLE
Improve environment variable extension for Java 17+

### LINKED ISSUE #771: Improve documentation of environment variable extension
From time to time (e.g. [AWS-Powertools-Lambda-Java-issue 1391](https://github.com/aws-powertools/powertools-lambda-java/issues/1391)) people think that the extension can't be uses with Java 17+, which is not true.

@beatngu13 suggested to somehow improve the documentation, to make this more visible.

### PR DESCRIPTION
**Pull request by beatngu13**
Closes #771 

Proposed commit message:

```
Improve environment variable extension for Java 17+ (#771 / #773)

This PR improves the environment variable extension for use with Java
17+. That is, better documentation and warnings as well as errors when
it comes to handling strong encapsulation of the JDK.

Closes: #771
PR: #773
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [ ] There is documentation (Javadoc and site documentation; added or updated)
* [ ] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [ ] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [ ] Site documentation in `.adoc` file references demo in `src/demo/java` instead of containing code blocks as text
* [ ] Only one sentence per line (especially in `.adoc` files)
* [ ] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [ ] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [ ] The `package-info.java` contains information about the new extension

Code (general)
* [ ] Code adheres to code style, naming conventions etc.
* [ ] Successful tests cover all changes
* [ ] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [ ] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Code (new package)
* [ ] The new package is exported in `module-info.java`
* [ ] The new package is also present in the tests
* [ ] The new package is opened for reflection to JUnit 5 in `module-info.java`
* [ ] The new package is listed in the contribution guide

Contributing
* [x] A prepared commit message exists
* [ ] The list of contributions inside `README.md` mentions the new contribution (real name optional)

### CODE REVIEW

File: src/main/java/org/junitpioneer/jupiter/EnvironmentVariableUtils.java [hunk 1]
```diff
@@ -100,8 +101,8 @@ private static Map<String, String> getFieldValue(Class<?> clazz, Object object,
 			field.setAccessible(true); //NOSONAR illegal access required to implement the extension
 		}
 		catch (InaccessibleObjectException ex) {
-			throw new ExtensionConfigurationException(
-				"Cannot access Java runtime internals to modify environment variables. "
+			throw new PreconditionViolationException(
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/jupiter/EnvironmentVariableUtils.java:104]: What's with line 66? What exact situation handles that `catch`? Should this be a `PreconditionViolationException` as well?

Comment (Bukama) [reply src/main/java/org/junitpioneer/jupiter/EnvironmentVariableUtils.java:104]: Yeah would keep them aligned.

**Review by Bukama — APPROVED** [review]
Nice changes to the docs and the warnings. As commented I am with you to change the exception line 66 too, to also give a hint there.

After you have aligned them, feel free to merge.

File: docs/environment-variables.adoc [hunk 2]
```diff
@@ -127,20 +127,49 @@ java.lang.reflect.InaccessibleObjectException: Unable to make field [...] access
...
-The next best thing is to allow access to that specific package:
+However, some tests require environment variables to be cleared, set, or restored.
+In this case, we recommend using `--add-opens` to grant JUnit Pioneer access to the aforementiond internals:
 
 [source]
 ----
+# to access java.lang.ProcessEnvironment.theEnvironment
 --add-opens java.base/java.util=$TARGET_MODULE
+# to access java.util.Collections$UnmodifiableMap.m
 --add-opens java.base/java.lang=$TARGET_MODULE
```
Comment (beatngu13) [inline docs/environment-variables.adoc:142]: Just realized that the comments should be swapped.

```suggestion
# to access java.util.Collections$UnmodifiableMap.m
--add-opens java.base/java.util=$TARGET_MODULE
# to access java.lang.ProcessEnvironment.theEnvironment
--add-opens java.base/java.lang=$TARGET_MODULE
```

**Pull request merged**